### ISSUE TITLE
Unable to finish the automated setup

### ISSUE THREAD

**Issue by ravenns** [schnoog/klippered#2]
Hello and thank you so much for providing this step-by-step tutorial.

The first time I followed the instructions, I managed to set up the Klipper, but I was unsure if I did all the steps regarding macros so I decided to start over. The second installation was not successful. Klipper was complaining that it could not connect to my 3D printer. So I tried several consecutive installations without success. And then I resorted to your script (wget "https://raw.githubusercontent.com/schnoog/klippered/main/install/Install_Ender3V3SE_Klipper.sh" && chmod +x Install_Ender3V3SE_Klipper.sh && ./Install_Ender3V3SE_Klipper.sh), which was not successful at all.

It starts all the steps and when it installs Klipper, the script begins endless "XXXXXXX....."  without ending. I left it all night and found a screen full of Xs the next day. Should the Ender 3 V3 SE be connected during the installation or not (my Ender was not connected).

Thanks in advance!

**Comment by schnoog** [comment]
Hi and sorry for the late reply.
The fast and dirty install script only really works on a new installation.
If there's already a config it unfortunately fails. 

I propose you do a manual installation following 
https://schnoog.eu/hobbies/3dprinting/ender-3-v3-se-klippered
starting with the section "The way to Klipper" so you can skip steps which are already in place.

**Comment by aamarques** [comment]
Hey, it Didn't work in a fresh install, though.
Rasp 4B 4GB
Rasp PI OS 64

**Comment by schnoog** [comment]
Hi, thanks for the feedback and sorry to hear that it didn't work.

I don't have a RPi4 to test and debug it on it, but I'll try to find some time to empty a rpi3 and install it there again.
Possibly something changed in the kiauh or klipper menuconfig which I missed.

**Comment by tdguchi** [comment]
Im having the same error, with same rpi, XXXXX for a long time till ctrl+c (we talk about almost 6 hours of XX) manual install works (the install, i cant make it work, my printer isnt detected, and even im using the display enabled firmware it stucks in a blue screen) Waiting for that fixed script.

thanks a lot!

edit: things that could be changed in kiauh, your manual install says "6, settings" but in kiauh now its "S, settings". Also everytime you launch kiauh it asks to use a new version. Maybe the trick is there...

**Comment by ravenns** [comment]
Yup, I also tried the manual installation as well as automated on RPi 4 - Ender 3 V3 SE not detected. I checked the repository of the 0x... user (sorry don't know the username by heart) and I think there is an issue with the commit there. This might be a reason for the Klipper to not work....

**Comment by aamarques** [comment]
Me too. The manual installation has some issues with MCU that do not work with your workarounds.
![image](https://github.com/user-attachments/assets/4ee89904-321a-4490-a69b-92dd4f1571fd)

**Comment by schnoog** [comment]
Based on that screenshot:
It seems your printer already has an older version of the klipper firmware installed.

If you did that on the same raspberry installation,  uninstall and reinstall klipper from within kiauh. 
That ensures that the package from jpcurti is used for compiling the MCU firmware 

Did the make command (Step 7) raise any error?
Did you give the resulting klipper.bin a new, unique (8.3) filename  
(klipper1.bin, klipper1234.bin and so on are all shortend to klippe~1.bin so you need to variate some of the first 6 letters of the bin file, I name them KMMDD.bin where MM is the month and DD the day, f.e. K0421.bin for my firmware compiles on April 21st)

**Comment by tdguchi** [comment]
> Based on that screenshot: It seems your printer already has an older version of the klipper firmware installed.
> 
> If you did that on the same raspberry installation, uninstall and reinstall klipper from within kiauh. That ensures that the package from jpcurti is used for compiling the MCU firmware
> 
> Did the make command (Step 7) raise any error? Did you give the resulting klipper.bin a new, unique (8.3) filename (klipper1.bin, klipper1234.bin and so on are all shortend to klippe~1.bin so you need to variate some of the first 6 letters of the bin file, I name them KMMDD.bin where MM is the month and DD the day, f.e. K0421.bin for my firmware compiles on April 21st)

Regarding that, I have fought with klipper, and after all, its working (now im levelling, another nightmare) but it only worked after removing the [e3v3se_display] thing in printer.cfg. Even with a compiled bin from the jpcurti repo. I dont know why.

Also, when printing, the nozzle cleaning (that line at the left) its out of the bed xD How can I fix it? which calibration its missed here?

thanks!

**Comment by schnoog** [comment]
The nozzle cleaning is most likely part of the start macro
 
This is how the relevant part in my start macro looks like

    G1 X2.1 Y20 Z0.3 F5000.0 ; Move to start position
    G1 X2.1 Y200.0 Z0.3 F1500.0 E15 ; Draw the first line
    G1 X3.8 Y200.0 Z0.3 F5000.0 ; Move to side a little
    G1 X3.8 Y20 Z0.3 F1500.0 E30 ; Draw the second line

X2.1 and X3.8 are the offset from left for the lines. 

Most probably you have similar lines in yours. Just increase the X values for the two lines so that they fit on the bed. 

If the klipper printer config was already in action you need (as far as I'm aware) to add the [e3v3se_display] line above the #*# <---------------------- SAVE_CONFIG ---------------------->  part.

**Comment by tdguchi** [comment]
thanks for that, i will try it later, im struggling with x y z level... and the the bed.

The display stuff was there, at the end of the file, at that time, no save_config thing. Mainsail said mcu is shutdown or something like that.
